modules/private.php

```diff
@@ -0,0 +1,307 @@
+ * Module Tags: Private
+ * Feature: Security
+ * Additional Search Queries: private, sandbox, launch, unlaunched, maintenance, coming soon
+ *
+ * @package Jetpack
+ */
+
+class Jetpack_Private {
+	static function init() {
+		add_action( 'parse_request', array( __CLASS__, 'privatize_blog' ), 100 );
+		add_action( 'login_init', array( __CLASS__, 'privatize_blog_maybe_mask_blog_name' ) );
+		add_filter( 'preprocess_comment', array( __CLASS__, 'privatize_blog_comments' ), 0 );
+		add_action( 'blog_privacy_selector', array( __CLASS__, 'privatize_blog_priv_selector' ) );
+		add_filter( 'robots_txt', array( __CLASS__, 'private_robots_txt' ) );
+		add_action( 'wp_head', array( __CLASS__, 'private_no_pinning' ) );
+		add_action( 'admin_init', array( __CLASS__, 'private_blog_prevent_requests' ), 9 );
+		add_action( 'check_ajax_referer', array( __CLASS__, 'private_blog_ajax_nonce_check' ), 9, 2 );
+		add_action( 'rest_pre_dispatch', array( __CLASS__, 'disable_rest_api' ) );
+		add_filter( 'jetpack_get_available_modules', array( __CLASS__, 'private_get_modules' ) );
+		add_action( 'update_option_blog_public', array( __CLASS__, 'private_update_option_blog_public' ) );
+		add_action( 'update_right_now_text', array( __CLASS__, 'add_private_dashboard_glance_items' ) );
+		add_action( 'jetpack_sync_before_send_queue_full_sync', array( __CLASS__, 'remove_privatize_blog_mask_blog_name_filter' ) );
+		add_action( 'jetpack_sync_before_send_queue_sync', array( __CLASS__, 'remove_privatize_blog_mask_blog_name_filter' ) );
+		add_action( 'opml_head', array( __CLASS__, 'hide_opml' ) );
+	}
+
+	/**
+	 * Returns the private site template for private blogs
+	 *
+	 * @param object $wp Current WordPress environment instance (passed by reference).
+	 */
+	static function privatize_blog( $wp ) {
+		global $pagenow;
+
+		if ( 'wp-login.php' === $pagenow ) {
+			return;
+		}
+
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			return;
+		}
+
+		// Serve robots.txt for private blogs.
+		if ( is_object( $wp ) && ! empty( $wp->query_vars['robots'] ) ) {
+			return;
+		}
+
+		if ( is_user_logged_in() && self::is_private_blog_user( get_current_blog_id() ) ) {
+			return;
+		}
+
+		include JETPACK__PLUGIN_DIR . '/modules/private/private.php';
+
+		exit;
+	}
+
+	/**
+	 * Does not check whether the blog is private. Accepts blog and user in various types.
+	 * Returns true for super admins.
+	 *
+	 * @param int   $blog Current WordPress blod id..
+	 */
+	protected static function is_private_blog_user( $blog ) {
+		if ( is_numeric( $blog ) ) {
+			$blog_id = intval( $blog );
+		} elseif ( is_object( $blog ) ) {
+			$blog_id = $blog->blog_id;
+		} elseif ( is_string( $blog ) ) {
+			$fields  = array(
+				'domain' => $blog,
+				'path'   => '/',
+			);
+			$blog    = get_blog_details( $fields );
+			$blog_id = $blog->blog_id;
+		} else {
+			$blog_id = get_current_blog_id();
+		}
+
+		/**
+		 * Filter the capabilites a user needs to have to see the site
+		 *
+		 * @module private sites
```

Comment:
This would need to be the module slug, `private`.

```suggestion
		 * @module private
```

Reply:
Done in [d35a9cd](https://github.com/Automattic/jetpack/pull/10333/commits/d35a9cd9fc38b795c7ecd09055df47d1f7f922bc)